src/main/java/net/neoforged/neoforge/transfer/handlers/templates/AggregateResourceHandler.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates;
+
+import net.neoforged.neoforge.transfer.IResource;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.handlers.IResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.IResourceHandlerModifiable;
+
+import java.util.stream.Stream;
+
+/**
+ * An {@link IResourceHandler} that aggregates multiple handlers into one.
+ * @param <T> The type of resource
+ */
+public class AggregateResourceHandler<T extends IResource> implements IResourceHandler<T> {
+    protected final IResourceHandler<T>[] handlers;
+
+    @SafeVarargs
+    public AggregateResourceHandler(IResourceHandler<T>... handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public int size() {
+        return Stream.of(getHandlers()).mapToInt(IResourceHandler::size).sum();
+    }
+
+    @Override
+    public T getResource(int index) {
+        for (IResourceHandler<T> storage : getHandlers()) {
+            if (index < storage.size()) {
```

Comment:
We probably only want to calculate `storage.size()` once here rather than doing it both here and when adjusting the index. Just in case, similar to this class' size implementation it isn't just a simple getter. On top of that we may want a couple private helper methods to simplify iterating over this and finding the correct index.